### DISCUSSION TITLE
What is the folder structure supposed to be for Koreader Vocab Builder imports?

### DISCUSSION THREAD

**Discussion by parthshahp** [FreeLanguageTools/vocabsieve#53] [General]
Title. I can't get the importer to work and curious as to what the folder structure should be. I have all of my books in the root directory of my Kobo.

**Comment by 1over137** [comment]
It doesn't have to be anything specific, just that it contains three things in any of its subdirectories:
1. your ebook files (actually, only metadata in the .sdr folders)
2. lookup_history.lua
3. vocabulary_builder.sqlite3


Note that context must be on for Vocab Builder, otherwise there will be no sentences.

Another common issue is badly labeled book language metadata. You can set the language in KOReader and it should work properly. VocabSieve filters out books in languages other than your target to avoid noise.

It recursively searches for these three things independently, so they don't have to be in any specific relative path to each other. Be sure to not make copies of them elsewhere in the device filesystem, otherwise it might cause your copy to be detected first and changes might not be reflected.
For Kobo, just select the root folder (I'm using a Kobo too). If you can't get it to work, you can send a minimal reproducible example containing one book and a number of words in the vocab builder.

Also read this: https://wiki.freelanguagetools.org/vocabsieve_reader_import
